oddt/toolkits/rdk.py

```diff
@@ -792,22 +795,32 @@ def _dicts(self):
             for residue in self.residues:
                 path = residue.Residue.GetSubstructMatch(aa)
                 if path:
-                    atom_dict['isbackbone'][np.array([residue.atommap[i] for i in path])] = True
-                    b.append((residue.MonomerInfo.GetResidueNumber(),
-                              residue.MonomerInfo.GetResidueName(),
+                    backbone_map = np.array([residue.atommap[i] for i in path])
+                    atom_dict['isbackbone'][backbone_map] = True
+                    b.append((residue.idx0,
+                              residue.number,
+                              residue.name,
                               conf.GetAtomPosition(residue.atommap[path[0]]),
                               conf.GetAtomPosition(residue.atommap[path[1]]),
                               conf.GetAtomPosition(residue.atommap[path[2]]),
                               conf.GetAtomPosition(residue.atommap[path[3]]),
                               False,
                               False))
+                    # set resid for atoms in atom_dict
+                    atom_dict['resid'][list(residue.atommap.values())] = residue.idx0
             res_dict = np.array(b, dtype=res_dtype)
             res_dict = detect_secondary_structure(res_dict)
-            atom_dict['isalpha'][np.in1d(atom_dict['resid'], res_dict[res_dict['isalpha']]['id'])] = True
-            atom_dict['isbeta'][np.in1d(atom_dict['resid'], res_dict[res_dict['isbeta']]['id'])] = True
+            # FIXME: this might be wrong if resnum is dubbled accross chains
```

Comment:
What about using `id` here, like in ob implementation? I think it's set properly and should be unique.

Reply:
Nice catch. Fixed that, and unified on OB's side.